src/hooks/posters.ts

```diff
@@ -75,17 +93,91 @@ export const usePosters = (posterHallId: string) => {
+      : posterVenues;
+    const bookmarkedFilterResultsCategory = bookmarkedFilter
+      ? liveFilterResultsCategory.filter(
+          (posterVenue) =>
+            //@ts-ignore
+            userPosterIds[posterVenue.id]?.[0] === posterVenue.id
+        )
+      : liveFilterResultsCategory;
+    return Array.from(
+      new Set(
+        bookmarkedFilterResultsCategory
+          .map((posterVenue) => posterVenue.poster?.categories)
+          .flat()
+      )
+    );
+  }, [liveFilter, posterVenues, bookmarkedFilter, userPosterIds]);
+
+  const filteredPostersByCategory = useMemo(
     () =>
-      liveFilter
-        ? posterVenues.filter((posterVenue) => posterVenue.isLive)
+      categoryFilter
+        ? posterVenues.filter((posterVenue) =>
+            posterVenue.poster?.categories.includes(categoryFilter)
+          )
         : posterVenues,
-    [posterVenues, liveFilter]
+    [posterVenues, categoryFilter]
   );
 
-  const [bookmarkedFilter, setBookmarkedFilter] = useState<boolean>(false);
-  const { userWithId } = useUser();
-  const userPosterIds = userWithId?.savedPosters ?? emptySavedPosters;
+  const subCategoryList = useMemo(() => {
+    const liveFilterResultsSubCategory = liveFilter
+      ? filteredPostersByCategory.filter((posterVenue) => posterVenue.isLive)
+      : filteredPostersByCategory;
+    const bookmarkedFilterResultsSubCategory = bookmarkedFilter
+      ? liveFilterResultsSubCategory.filter(
+          (posterVenue) =>
+            //@ts-ignore
+            userPosterIds[posterVenue.id]?.[0] === posterVenue.id
+        )
+      : liveFilterResultsSubCategory;
+    return categoryFilter
+      ? Array.from(
+          new Set(
+            bookmarkedFilterResultsSubCategory
+              .map((posterVenue) => posterVenue.poster?.subcategories)
+              .flat()
+              .filter((subcategory) =>
+                subcategory?.category.includes(categoryFilter)
+              )
+              .map((subcategory) => subcategory?.subcategory)
+              .flat()
+          )
+        ).filter(isTruthy)
+      : [];
+  }, [
+    liveFilter,
+    filteredPostersByCategory,
+    bookmarkedFilter,
+    categoryFilter,
+    userPosterIds,
+  ]);
+
+  const filteredPostersBySubCategory = useMemo(
+    () =>
+      subCategoryFilter
+        ? filteredPostersByCategory.filter((posterVenue) =>
+            posterVenue.poster?.subcategories
+              ?.map((subcategory) => subcategory?.subcategory)
+              .includes(subCategoryFilter)
+          )
+        : filteredPostersByCategory,
+    [filteredPostersByCategory, subCategoryFilter]
+  );
+
+  const liveFilteredPosterVenues = useMemo(
+    () =>
+      liveFilter
+        ? filteredPostersBySubCategory.filter(
+            (filteredPostersBySubCategory) =>
+              filteredPostersBySubCategory.isLive
+          )
+        : filteredPostersBySubCategory,
+    [filteredPostersBySubCategory, liveFilter]
+  );
```

Comment:
There is already filtering going based on 'live' above this in `liveFilterResultsSubCategory` and similar. Why is this also needed here?

Reply:
Additionally, use the guard pattern as described above rather than nesting complex ternaries like this.